File: subprojects/dumbbell-core/src/main/java/com/tambapps/marcel/dumbbell/DumbbellException.java
package com.tambapps.marcel.dumbbell;

public class DumbbellException extends RuntimeException {

  public DumbbellException(String message) {
    super(message);
  }

  public DumbbellException(String message, Throwable cause) {
    super(message, cause);
  }

  public DumbbellException(Throwable cause) {
    super(cause != null ? cause.getMessage() : null, cause);
  }
}

File: marcel-subprojects/dumbbell-core/src/main/kotlin/com/tambapps/marcel/dumbbell/DumbbellException.kt
package com.tambapps.marcel.dumbbell

class DumbbellException : RuntimeException {
  constructor(message: String) : super(message)

  constructor(message: String, cause: Throwable) : super(message, cause)

  constructor(cause: Throwable) : super(cause.message, cause)
}
